src/features/ProfilePage/components/PublicInfo.tsx

```diff
@@ -1,35 +1,88 @@
+import { useDebounce } from '@/hooks/useDebounce';
+import { useUpdateMentorMutation } from '@/features/MentorPage/mentorPageApi';
+import { useUpdateUserMutation } from '@/features/Authentication/authenticationApi';
+
+import {
+  DEFAULT_ICON_SIZE,
+  palette,
+  SAVE_DELAY_MS,
+} from '@/components/constants';
 import LabeledInput from '@/components/LabeledInput';
 import Slider from '@/components/Slider';
 import Text from '@/components/Text';
 import TextInput from '@/components/TextInput';
 
+import type {
+  MentorUser,
+  User,
+} from '@/features/Authentication/authenticationApi';
+
 const PublicInfo = () => {
   const { t } = useTranslation('profile');
-  const userInfo = useAppSelector(selectUserInfo);
-
-  const [displayName, setDisplayName] = useState('');
-  const [birthYear, setBirthYear] = useState('');
-  const [area, setArea] = useState('');
-  const [status, setStatus] = useState('');
-  const [isActive, setIsActive] = useState(true);
-  const [story, setStory] = useState('');
+  const user = useAppSelector(selectUser);
+  const mentor = useAppSelector(selectMentor);
+  const [updateUser] = useUpdateUserMutation();
+  const [updateMentor] = useUpdateMentorMutation();
+
+  const [birthYear, setBirthYear] = useState(String(mentor.birth_year));
+  const [displayName, setDisplayName] = useState(mentor.display_name);
+  const [isAbsent, setIsAbsent] = useState(mentor.is_vacationing);
+  const [region, setRegion] = useState(mentor.region);
+  const [status, setStatus] = useState(mentor.status_message);
+  const [story, setStory] = useState(mentor.story);
   const [topicSearchValue, setTopicSearchValue] = useState('');
 
-  const toggleIsActive = () => setIsActive(isActive => !isActive);
+  const saveMentorData = useDebounce(
+    (mentor: MentorUser) => updateMentor(mentor),
+    SAVE_DELAY_MS,
+  );
+
+  const saveUserAndMentorData = useDebounce(
+    (mentor: MentorUser, user: User) => {
+      updateUser(user);
+      updateMentor(mentor);
+    },
+    SAVE_DELAY_MS,
+  );
 
-  useEffect(() => {
-    const { displayName, active } = userInfo;
-    if (displayName !== null) setDisplayName(displayName);
-    if (isActive !== null) setIsActive(active);
-  }, [userInfo]);
+  const handleDisplayNameChange = (updatedDisplayName: string) => {
+    setDisplayName(updatedDisplayName);
+    saveUserAndMentorData(
+      { ...mentor, display_name: updatedDisplayName },
+      { ...user, display_name: updatedDisplayName },
+    );
+  };
+
+  const handleBirthYearChange = (updatedBirthYear: string) => {
+    setBirthYear(updatedBirthYear);
+    saveMentorData({ ...mentor, birth_year: Number(updatedBirthYear) });
+  };
+
+  const handleRegionChange = (updatedRegion: string) => {
+    setRegion(updatedRegion);
+    saveMentorData({ ...mentor, region: updatedRegion });
+  };
+
+  const handleStatusChange = (updatedStatus: string) => {
+    setStatus(updatedStatus);
+    saveMentorData({ ...mentor, status_message: updatedStatus });
+  };
```

Comment:
Usually i like to create an abstraction instead of doing multiple separate handlers.
Like 

```typescript
  const [localData, setLocalData] = useState<MentorUser>(mentor)

  const updateMentorData = (key: keyof MentorUser, value: string) => {
    setLocalData({...localData, [key]: value});
    saveMentorData({ ...mentor, [key]: value }); // this is the debounced func
}
```

That means we dont have to have so many separate useState's but combine them in one object.
It also means that maybe there needs to be done some jujitsu for the typings to go correctly.

This is also my personal opinion of making the codebase a bit clearer and remove repetition

Reply:
Here is provided some typescriptjujitsu: 

```typescript
  const updateMentorData = <K extends keyof MentorUser>(
    key: K,
    value: MentorUser[K],
  ) => {
    setLocalData({ ...localData, [key]: value });
    saveMentorData({ ...mentor, [key]: value });
  };
  ```

Reply:
<img width="1364" alt="image" src="https://github.com/user-attachments/assets/615750bb-c876-4c83-b947-c3b6b4199f4f">

Reply:
I implemented some of this , but everything cannot be refactored to use one function because the handlers are different. But I think it's good enough for now.

Reply:
It could be done actually. Just return the data we want to update. I update my comment in the review again